packages/react-native/React/Views/RCTViewManager.m

```diff
@@ -336,7 +346,17 @@ - (void)updateAccessibilityTraitsForRole:(RCTView *)view withDefaultView:(RCTVie
       : view.reactAccessibilityElement.accessibilityRole ? view.reactAccessibilityElement.accessibilityRoleTraits
                                                          : (defaultView.accessibilityTraits & AccessibilityRolesMask);                        
 }
-#endif // [macOS]
+#else // [macOS
+- (void) updateAccessibilityRole:(RCTView *)view withDefaultView:(RCTView *)defaultView
+{
+  // First check the value from `role`
+  view.reactAccessibilityElement.accessibilityRole = view.reactAccessibilityElement.role ? view.reactAccessibilityElement.role : nil;
+  // Fallback to `accessibilityRole` if nil, or the defaultView's NSAccessibilityRole
+  if (view.reactAccessibilityElement.accessibilityRole == nil) {
+    view.reactAccessibilityElement.accessibilityRole = view.reactAccessibilityElement.accessibilityRoleInternal ? view.reactAccessibilityElement.accessibilityRoleInternal : defaultView.accessibilityRole;
```

Comment:
OK, you mean this defaultView. TBH, this feels like unreachable code and the equivalent of setting to nil. I guess my preference is to copy whatever iOS does? And/or check if Fabric does something different ( I think the file is `packages/react-native/ReactCommon/react/renderer/components/view/accessibilityPropsConversions.h`)

Reply:
iOS maps this to UIAccessibilityTraitNone... so the question is should we map to NSAccessibilityRoleUnknown? nil? Group?  All of those have slightly different implications on the accessibility representation, I would guess.

Reply:
I updated it to nil in the latest iteration, which I think may be the most accurate representation?

Reply:
Yeah I looked around Fabric and saw that iOS has `none` while macOS doesn't.. I think nil is probably most accurate.